magicctapipe/scripts/lst1_magic/muon_analysis_LST_or_MAGIC_data.py

```diff
@@ -0,0 +1,169 @@
+    output_dir:
+    config:
+    process_run:
+
+    """
+
+    event_source = MAGICEventSource(input_url=input_file)
+    subarray = event_source.subarray
+    obs_id = event_source.obs_ids[0]
+    tel_id = event_source.telescope
+
+    # Create the table which will contain the selected muon ring parameters
+    muon_parameters = create_muon_table()
+    muon_parameters['telescope_name'] = []
+
+    logger.info(f'\nProcess the following data (process_run = {process_run}):')
+
+    # Configure the MAGIC cleaning:
+    config_cleaning = config['MAGIC']['magic_clean']
+
+    if config_cleaning['find_hotpixels'] == 'auto':
+        config_cleaning.update({'find_hotpixels': True})
+
+    logger.info('\nConfiguration for the image cleaning:')
+    logger.info(config_cleaning)
+
+    ped_type = config_cleaning.pop('pedestal_type')
+    i_ped_type = np.where(np.array(pedestal_types) == ped_type)[0][0]
+
+    camera_geom = event_source.subarray.tel[tel_id].camera.geometry
+    magic_clean = MAGICClean(camera_geom, config_cleaning)
+
+    # Prepare for saving muons data to an output file:
+    Path(output_dir).mkdir(exist_ok=True, parents=True)
+
+    if process_run:
+        output_file = f'{output_dir}/muons_M{tel_id}.Run{obs_id:08}.fits'
+    else:
+        subrun_id = event_source.metadata['subrun_number']
+        output_file = f'{output_dir}/muons_M{tel_id}.Run{obs_id:08}.{subrun_id:03}.fits'
+
+    # Start processing events:
+    logger.info('\nProcessing the events:')
+
+    # We give None as calibrator for the muons analysis as we work with extracted charges
+    r1_dl1_calibrator_for_muon_rings_magic = None
+
+    # load muon analysis config
+    muon_config = {}
+    if 'muon_ring' in config['MAGIC']:
+        muon_config = config['MAGIC']['muon_ring']
+    # Select the telescope name to be filed in muon_parameters['telescope_name']
+    if tel_id == 1:
+        tel_name = f'MAGIC-I'
+    elif tel_id == 2:
+        tel_name = f'MAGIC-II'
+    else:
+        tel_name = f'MAGIC_?'
+
+    for event in event_source:
+
+        # Apply the image cleaning:
+        dead_pixels = event.mon.tel[tel_id].pixel_status.hardware_failing_pixels[0]
+        try:
+            badrms_pixels = event.mon.tel[tel_id].pixel_status.pedestal_failing_pixels[i_ped_type]
+        except TypeError:
+            badrms_pixels = np.zeros(dead_pixels.shape, dtype=bool)
+        unsuitable_mask = np.logical_or(dead_pixels, badrms_pixels)
+
+        signal_pixels, image, peak_time = magic_clean.clean_image(
+            event.dl1.tel[tel_id].image, event.dl1.tel[tel_id].peak_time, unsuitable_mask,
+        )
+        perform_muon_analysis(muon_parameters,
+                              event=event,
+                              telescope_id=tel_id,
+                              telescope_name=tel_name,
+                              image=image,
+                              subarray=subarray,
+                              r1_dl1_calibrator_for_muon_rings=
```

Comment:
I think putting here directly None except of a separate variable set to None would be more clear